### DISCUSSION TITLE
Is it planned to upgrade for typer 0.12 ?

### DISCUSSION THREAD

**Discussion by Paillat-dev** [explosion/spaCy#13464] [Help: Installation] [answered]
I'm currently encountering a dependency conflict in my project involving the Typer package. spaCy supports Typer versions up to 0.10, but I am also using Gradio, which as of their latest updates, requires Typer 0.12 or newer (reference: [Gradio Issue #8072](https://github.com/gradio-app/gradio/issues/8072#issue-2252289370)).

I understand that installing Typer==0.9[all] is a possible workaround, but this solution does not align with the dependency locking and resolution strategies I'm employing. Overriding dependencies generally leads to less stable and maintainable setups, especially when using tools like pdm.

Is the team considering updating spaCy's requirements to support newer versions of Typer? This change would help maintain compatibility with other libraries that depend on newer versions of Typer and facilitate more robust project environments.

Thank you for considering this upgrade.

**Comment by svlandeg** [comment, marked as answer]
Hi, thanks for the report! Yes - we'll get that Typer pin updated in the near future.

**Reply by Paillat-dev** [reply]
Thanks much!

**Reply by svlandeg** [reply]
PR: https://github.com/explosion/spaCy/pull/13471

**Reply by svlandeg** [reply]
It took a while because we ran into an unexpected issue with the wheel building process, but v3.7.5 is now released on PyPI: https://pypi.org/project/spacy/3.7.5/ ✅

**Comment by nokados** [comment]
It's also needed to upgrade _typer_ version in weasel - https://github.com/explosion/weasel/pull/87

**Reply by aymenkrifa** [reply]
https://github.com/explosion/weasel/pull/87 is now merged :white_check_mark: 

[edit: Weasel 0.4.1 (containing the `typer` upgrade is released]
